modules/alloy/src/main/ts/ephox/alloy/dragging/snap/Presnaps.ts

```diff
@@ -1,17 +1,18 @@
-import { Option } from '@ephox/katamari';
-import { Attr, Position } from '@ephox/sugar';
+import { Option, Options } from '@ephox/katamari';
+import { Attr, Element, Position } from '@ephox/sugar';
 
 import { AlloyComponent } from '../../api/component/ComponentApi';
 import { SnapsConfig } from '../common/DraggingTypes';
 
+const parseAttrToInt = (element: Element, name: string) =>
+  Attr.getOpt(element, name).map((value) => parseInt(value, 10)).filter((value) => !isNaN(value));
```

Comment:
The types won't allow `parseInt` to accept `undefined` so I've had to do a check and just make it return `NaN` directly instead.

Reply:
For reference: https://github.com/microsoft/TypeScript/blob/6d7539a2110b5b9edd723bd8fc7ecc9b4be69943/src/lib/es2015.core.d.ts#L265 and https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt so while it may work, it's more likely by coincidence than intention.

Reply:
> https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt

`If this argument is not a string, then it is converted to one using the ToString abstract operation`

So the TypeScript types are wrong (which is fine, nobody should be passing `undefined` to `parseInt`, but it is unhelpful in this scenario)

Reply:
Good catch, I missed that sorry!